core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -116,8 +117,9 @@ static TableMetadata newTableMetadata(Schema schema,
 
     return new TableMetadata(null, formatVersion, UUID.randomUUID().toString(), location,
         INITIAL_SEQUENCE_NUMBER, System.currentTimeMillis(),
-        lastColumnId.get(), freshSchema, INITIAL_SPEC_ID, ImmutableList.of(freshSpec),
-        freshSpec.lastAssignedFieldId(), freshSortOrderId, ImmutableList.of(freshSortOrder),
+        lastColumnId.get(), freshSchema.schemaId(), ImmutableList.of(freshSchema),
+        INITIAL_SPEC_ID, ImmutableList.of(freshSpec), freshSpec.lastAssignedFieldId(),
```

Comment:
I like that you use `freshSchema.schemaId()` instead of using `INITIAL_SCHEMA_ID` again. Should we do the same for `INITIAL_SPEC_ID`?